Make CRS equality looser

omad opened the pull request:
### Reason for this pull request
Issue #243 found an issue comparing two different representations of an Australian Albers CRS Projection. This PR makes a change so that two `CRS` objects are equal if either a _Proj.4_ based comparison or a _GDAL_ based comparison says they are equal.



### Proposed changes
- Trust either GDAL or Proj4 based comparison of CRS
- Add test for broken CRS comparison

 - [x] Closes issue #243
 - [x] Tests added / passed
 - [x] Fully documented, including `docs/about/whats_new.rst` for all changes

omad committed:
Make CRS equality looser

Trust either a _Proj.4_ based comparison  or a _GDAL_ based comparison. 

Closes #243

omad committed:
Merge branch 'develop' into fix-crs-comparison

omad committed:
Merge branch 'develop' into fix-crs-comparison

Kirill888 commented on datacube/utils/geometry.py:
```diff
@@ -219,8 +219,12 @@ def __repr__(self):
             other = CRS(other)
-        canonical = lambda crs: set(crs.ExportToProj4().split() + ['+wktext'])
-        return canonical(self._crs) == canonical(other._crs)  # pylint: disable=protected-access
+        gdal_thinks_issame = self._crs.IsSame(other._crs) == 1  # pylint: disable=protected-access
```
why not early exit?

``` python
if self._crs.IsSame(other._crs) == 1:  # trust gdal assesment
    return True
```

omad replied:
No particular reason, just trying to be self documenting.

Kirill888 replied:
It all depends on how costly is computation and how often it is done, no point computing the second one if the first one is `True`. Building a string, splitting it into an array and constructing a set out of that, times 2 is a bit of work, but I guess it isn't being done a lot.

Kirill888 reviewed (dismissed):
I guess/hope we are not comparing CRSs in bulk?

omad committed:
Slight optimisation to CRS comparison

Return early if GDAL thinks they're the same.

ps. if only python was more lazily evaluated... :)
